### PR TITLE
set sum of uncompressed column size as row group size for parquet files

### LINKED ISSUE #3530: Incorrect row group total_byte_size written to parquet file
**Describe the bug**
The row group total_byte_size currently written to the parquet file is the compressed size and not the uncompressed size as expected

**To Reproduce**
For example uk_cities_with_headers.csv converted to parquet with schema
```
message uk_cities {
  required binary city (STRING);
  required double lat;
  required double lng;
}
```
shows the following stats
```
$ java -jar parquet-tools-1.10.99.7.2.15.0-147.jar meta uk_cities_rust.parquet
....
row group 1: RC:37 TS:1595 OFFSET:4
--------------------------------------------------------------------------------
city:         BINARY SNAPPY DO:4 FPO:710 SZ:815/1115/1.37 VC:37 ENC:PLAIN,RLE_DICTIONARY,RLE ST:[min: Aberdeen, Aberdeen City, UK, max: Worthing, West Sussex, UK, num_nulls not defined]
lat:          DOUBLE SNAPPY DO:907 FPO:1224 SZ:390/383/0.98 VC:37 ENC:PLAIN,RLE_DICTIONARY,RLE ST:[min: 50.376289, max: 57.653484, num_nulls not defined]
lng:          DOUBLE SNAPPY DO:1349 FPO:1666 SZ:390/383/0.98 VC:37 ENC:PLAIN,RLE_DICTIONARY,RLE ST:[min: -7.318268, max: 0.573453, num_nulls not defined]
```

**Expected behavior**
The total size is expected to be sum of the uncompressed column sizes 1115 + 383 + 383 = 1881 and not compressed size 815 + 390 + 390 = 1595

**Additional context**
Same csv converted to parquet using python pyarrow shows
```
java -jar parquet-tools-1.10.99.7.2.15.0-147.jar meta uk_cities_py.parquet
row group 1: RC:37 TS:1945 OFFSET:4
--------------------------------------------------------------------------------
city:         BINARY SNAPPY DO:4 FPO:713 SZ:826/1123/1.36 VC:37 ENC:RLE,PLAIN,RLE_DICTIONARY ST:[min: Aberdeen, Aberdeen City, UK, max: Worthing, West Sussex, UK, num_nulls: 0]
lat:          DOUBLE SNAPPY DO:941 FPO:1258 SZ:418/411/0.98 VC:37 ENC:RLE,PLAIN,RLE_DICTIONARY ST:[min: 50.376289, max: 57.653484, num_nulls: 0]
lng:          DOUBLE SNAPPY DO:1454 FPO:1771 SZ:418/411/0.98 VC:37 ENC:RLE,PLAIN,RLE_DICTIONARY ST:[min: -7.318268, max: 0.573453, num_nulls: 0]
```
Here the total size matches the columns uncompressed size
1945 = 1123 + 411 + 411

### PR DESCRIPTION
**Pull request by sidred**
# Which issue does this PR close?

Closes #3530 

# Rationale for this change
 
Fix the row group total_byte_size of the parquet format. This is currently set to the compressed size. As per the spec https://github.com/apache/parquet-format/blob/613a1cf4475c662457a0fd81a894ce4709799e3b/src/main/thrift/parquet.thrift#L814
this should be the uncompressed size

# What changes are included in this PR?
update the parquet file row group to sum of uncompressed column sizes

# Are there any user-facing changes?

The row group size of parquet file will change and correctly reflect the sum of the uncompressed column sizes

### CODE REVIEW

File: parquet/src/file/writer.rs [hunk 1]
```diff
@@ -499,9 +499,11 @@ impl<'a, W: Write> SerializedRowGroupWriter<'a, W> {
             self.assert_previous_writer_closed()?;
 
             let column_chunks = std::mem::take(&mut self.column_chunks);
+            let total_bytes_size =
+                column_chunks.iter().map(|c| c.uncompressed_size()).sum();
             let row_group_metadata = RowGroupMetaData::builder(self.descr.clone())
                 .set_column_metadata(column_chunks)
-                .set_total_byte_size(self.total_bytes_written as i64)
+                .set_total_byte_size(total_bytes_size)
```
Comment (sidred) [inline parquet/src/file/writer.rs:506]: I am not sure this is the right way. One alternative is to set the self.total_bytes_written to `*total_bytes_written += r.metadata.uncompressed_size();` in the on_close closure

**Review by tustvold — COMMENTED** [review]
Would it be possible to get a test of this, mainly so we don't accidentally break it in future. Perhaps as part of the layout integration test?

**Comment by sidred** [comment]
I have added a test but seems to pass both old and new versions. It will be great if you could write it out.

Also i have changed the implementation to track the written columns only (as the total written bytes does). So if any columns are skipped, we still write the correct size.

**Comment by tustvold** [comment]
> It will be great if you could help here

Compression isn't enabled by default, perhaps you need to enable it for that test? Otherwise I can take a look tomorrow

**Comment by sidred** [comment]
I have added compression tests to boolean and int writer tests. These tests fail on current master.

**Review by tustvold — APPROVED** [review]
LGTM thank you, it looks like some unrelated files made it into this PR, could you possibly remove them and I can then get this one in :smile:

**Pull request merged**